rero_ils/modules/documents/serializers.py

```diff
@@ -166,8 +179,447 @@ def _process_library_buckets(cls, org, lib_buckets):
+            query_es = sru.get('query_es')
+            next_record = start_record + maximum_records
+            root = element.searchRetrieveResponse()
+            root.append(element.version('1.1'))
+            root.append(element.numberOfRecords(str(number_of_records)))
+            if next_record > 1 and next_record < number_of_records:
+                root.append(element.nextRecordPosition(str(next_record)))
+            data = element.records()
+            for record in records:
+                data.append(dump_record(record))
+            root.append(data)
+            echoed_search_rr = element.echoedSearchRetrieveRequest()
+            if query:
+                echoed_search_rr.append(element.query(query))
+            if query_es:
+                echoed_search_rr.append(element.query_es(query_es))
+            if start_record:
+                echoed_search_rr.append(
+                    element.startRecord(str(start_record)))
+            if maximum_records:
+                echoed_search_rr.append(
+                    element.maximumRecords(str(maximum_records)))
+            echoed_search_rr.append(element.recordPacking('XML'))
+            echoed_search_rr.append(
+                element.recordSchema(
+                    'info:sru/schema/1/marcxml-v1.1-light'))
+            echoed_search_rr.append(element.resultSetTTL('0'))
+            root.append(echoed_search_rr)
+
+        if xslt_filename is not None:
+            xslt_root = etree.parse(open(xslt_filename))
+            transform = etree.XSLT(xslt_root)
+            root = transform(root).getroot()
+
+        return root
+
+    def dumps(self, total, records, sru, xslt_filename=None, **kwargs):
+        """Dump records into a MarcXMLSRU file."""
+        root = self.dumps_etree(
+            total=total,
+            records=records,
+            sru=sru,
+            xslt_filename=xslt_filename
+        )
+        return etree.tostring(
+            root,
+            pretty_print=True,
+            xml_declaration=True,
+            encoding='UTF-8',
+            **kwargs
+        )
+
+    def serialize_search(self, pid_fetcher, search_result,
+                         item_links_factory=None, **kwargs):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param item_links_factory: Factory function for the items in result.
+            (Default: ``None``)
+        :returns: The objects serialized.
+        """
+        language = request.args.get('ln', DEFAULT_LANGUAGE)
+        records = self.transform_records(
+            hits=search_result['hits']['hits'],
+            pid_fetcher=pid_fetcher,
+            language=language,
+            item_links_factory=item_links_factory
+        )
+        return self.dumps(
+            total=search_result['hits']['total'],
+            sru=search_result['hits']['total'].get('sru', {}),
```

Comment:
How is it possible that the sru information is on the elasticsearch total field?

Reply:
Moved it directly under `hits`